{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Union of all regs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pybedtools as pybed\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "library(ActiveDriverWGS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>134901</td>\n",
       "      <td>134902</td>\n",
       "      <td>AL627309.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>367640</td>\n",
       "      <td>367641</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>621059</td>\n",
       "      <td>621060</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>738532</td>\n",
       "      <td>738533</td>\n",
       "      <td>AL669831.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>818043</td>\n",
       "      <td>818044</td>\n",
       "      <td>AL645608.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20308</th>\n",
       "      <td>chr21</td>\n",
       "      <td>47720095</td>\n",
       "      <td>47720096</td>\n",
       "      <td>C21orf58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20309</th>\n",
       "      <td>chr21</td>\n",
       "      <td>47744036</td>\n",
       "      <td>47744037</td>\n",
       "      <td>PCNT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20310</th>\n",
       "      <td>chr21</td>\n",
       "      <td>47878812</td>\n",
       "      <td>47878813</td>\n",
       "      <td>DIP2A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20311</th>\n",
       "      <td>chr21</td>\n",
       "      <td>48018875</td>\n",
       "      <td>48018876</td>\n",
       "      <td>S100B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20312</th>\n",
       "      <td>chr21</td>\n",
       "      <td>48055079</td>\n",
       "      <td>48055080</td>\n",
       "      <td>PRMT2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20313 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr     start       end        gene\n",
       "0       chr1    134901    134902  AL627309.1\n",
       "1       chr1    367640    367641      OR4F29\n",
       "2       chr1    621059    621060      OR4F16\n",
       "3       chr1    738532    738533  AL669831.1\n",
       "4       chr1    818043    818044  AL645608.2\n",
       "...      ...       ...       ...         ...\n",
       "20308  chr21  47720095  47720096    C21orf58\n",
       "20309  chr21  47744036  47744037        PCNT\n",
       "20310  chr21  47878812  47878813       DIP2A\n",
       "20311  chr21  48018875  48018876       S100B\n",
       "20312  chr21  48055079  48055080       PRMT2\n",
       "\n",
       "[20313 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.8/subprocess.py:853: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "\n",
    "####\n",
    "#Function for calling promoters\n",
    "####\n",
    "#awk '{print $0}' /index/hg19/Homo_sapiens.GRCh37.87.chr.gtf | grep \"protein_coding\" | awk '{if ($3 == \"gene\") print \"chr\"$1\"\\t\"$4\"\\t\"$5\"\\t\"$7\"\\t\"$14 }' > /home/darragh/Annotation_reg_elements/hg19_protein_coding_minimal.gtf\n",
    "gtf = \"/home/darragh/Annotation_reg_elements/hg19_protein_coding_minimal.gtf\"\n",
    "def Identify_coding_promoters(gtf):\n",
    "    hg19_gtf = pd.read_table(gtf)\n",
    "    hg19_gtf.columns = [\"chr\",\"start\",\"end\", \"strand\", \"gene\"]\n",
    "    hg19_gtf = hg19_gtf[[\"chr\",\"start\",\"end\", \"gene\"]]\n",
    "    seqnames_we_like = [\"chr1\",\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\", \"chrY\"]\n",
    "    hg19_gtf = hg19_gtf[hg19_gtf['chr'].isin(seqnames_we_like)]\n",
    "    hg19_gtf['end'] = hg19_gtf['start']+1\n",
    "    hg19_gtf['gene'] = hg19_gtf['gene'].str.rstrip(';')\n",
    "\n",
    "    return(hg19_gtf)\n",
    "\n",
    "Coding_promoter_df = Identify_coding_promoters(gtf)\n",
    "display(Coding_promoter_df)\n",
    "Coding_promoter = pybed.BedTool.from_dataframe(Coding_promoter_df)\n",
    "Coding_promoter = Coding_promoter.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.8/subprocess.py:853: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n",
      "/usr/lib/python3.8/subprocess.py:853: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>9909</td>\n",
       "      <td>11509</td>\n",
       "      <td>chr1_9909_11509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13491</td>\n",
       "      <td>14174</td>\n",
       "      <td>chr1_13491_14174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28115</td>\n",
       "      <td>30038</td>\n",
       "      <td>chr1_28115_30038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>34333</td>\n",
       "      <td>35501</td>\n",
       "      <td>chr1_34333_35501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>38542</td>\n",
       "      <td>39129</td>\n",
       "      <td>chr1_38542_39129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99069</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59099066</td>\n",
       "      <td>59101436</td>\n",
       "      <td>chrY_59099066_59101436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99070</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59213304</td>\n",
       "      <td>59214942</td>\n",
       "      <td>chrY_59213304_59214942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99071</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59220433</td>\n",
       "      <td>59221668</td>\n",
       "      <td>chrY_59220433_59221668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99072</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59348730</td>\n",
       "      <td>59349716</td>\n",
       "      <td>chrY_59348730_59349716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99073</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59362785</td>\n",
       "      <td>59363497</td>\n",
       "      <td>chrY_59362785_59363497</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99074 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr     start       end                      id\n",
       "0      chr1      9909     11509         chr1_9909_11509\n",
       "1      chr1     13491     14174        chr1_13491_14174\n",
       "2      chr1     28115     30038        chr1_28115_30038\n",
       "3      chr1     34333     35501        chr1_34333_35501\n",
       "4      chr1     38542     39129        chr1_38542_39129\n",
       "...     ...       ...       ...                     ...\n",
       "99069  chrY  59099066  59101436  chrY_59099066_59101436\n",
       "99070  chrY  59213304  59214942  chrY_59213304_59214942\n",
       "99071  chrY  59220433  59221668  chrY_59220433_59221668\n",
       "99072  chrY  59348730  59349716  chrY_59348730_59349716\n",
       "99073  chrY  59362785  59363497  chrY_59362785_59363497\n",
       "\n",
       "[99074 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "HMEC = pybed.BedTool.from_dataframe(pd.read_table(\"/home/darragh/New/Regulatory_elements/HMEC_reg_elements.txt\", sep=\"\\t\",header=None))\n",
    "MCF10A = pybed.BedTool.from_dataframe(pd.read_table(\"/home/darragh/New/Regulatory_elements/MCF10A_reg_elements.txt\", sep=\"\\t\",header=None))\n",
    "MCF7 = pybed.BedTool.from_dataframe(pd.read_table(\"/home/darragh/New/Regulatory_elements/MCF7_reg_elements.txt\", sep=\"\\t\",header=None))\n",
    "T47D = pybed.BedTool.from_dataframe(pd.read_table(\"/home/darragh/New/Regulatory_elements/T47D_reg_elements.txt\", sep=\"\\t\",header=None))\n",
    "MDAMB231 = pybed.BedTool.from_dataframe(pd.read_table(\"/home/darragh/New/Regulatory_elements/MDAMB231_reg_elements.txt\", sep=\"\\t\",header=None))\n",
    "MDAMB436 = pybed.BedTool.from_dataframe(pd.read_table(\"/home/darragh/New/Regulatory_elements/MDAMB436_reg_elements.txt\", sep=\"\\t\",header=None))\n",
    "MDAMB468 = pybed.BedTool.from_dataframe(pd.read_table(\"/home/darragh/New/Regulatory_elements/MDAMB468_reg_elements.txt\", sep=\"\\t\",header=None))\n",
    "BT549 = pybed.BedTool.from_dataframe(pd.read_table(\"/home/darragh/New/Regulatory_elements/BT549_reg_elements.txt\", sep=\"\\t\",header=None))\n",
    "\n",
    "All = HMEC.cat(MCF10A, postmerge=False)\n",
    "All = All.cat(MCF7, postmerge=False)\n",
    "All = All.cat(T47D, postmerge=False)\n",
    "All = All.cat(MDAMB231, postmerge=False)\n",
    "All = All.cat(MDAMB436, postmerge=False)\n",
    "All = All.cat(MDAMB468, postmerge=False)\n",
    "All = All.cat(BT549, postmerge=False)\n",
    "\n",
    "All = All.sort()\n",
    "All = All.merge()\n",
    "\n",
    "All = pd.read_table(All.fn, header=None)\n",
    "All.columns = ['chr', 'start', 'end']\n",
    "All['id'] = All['chr'] + '_' + All['start'].astype(str) + '_' + All['end'].astype(str)\n",
    "All.to_csv(\"/home/darragh/New/ActiveDriverWGS/All_regs.txt\", sep=\"\\t\", header=False, index=False)\n",
    "\n",
    "\n",
    "display(All)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2606844/1692989507.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ER_active_promoters['annotation'] = \"ER_active_promoter\"\n",
      "/tmp/ipykernel_2606844/1692989507.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ER_polycomb_promoters['annotation'] = \"ER_polycomb_promoter\"\n",
      "/usr/lib/python3.8/subprocess.py:853: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n",
      "/usr/lib/python3.8/subprocess.py:853: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>21750</th>\n",
       "      <td>chr1</td>\n",
       "      <td>858733</td>\n",
       "      <td>861980</td>\n",
       "      <td>chr1_858733_861980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21751</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1049470</td>\n",
       "      <td>1052472</td>\n",
       "      <td>chr1_1049470_1052472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21752</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10490032</td>\n",
       "      <td>10491779</td>\n",
       "      <td>chr1_10490032_10491779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21753</th>\n",
       "      <td>chr1</td>\n",
       "      <td>197169303</td>\n",
       "      <td>197171676</td>\n",
       "      <td>chr1_197169303_197171676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21754</th>\n",
       "      <td>chr1</td>\n",
       "      <td>197743251</td>\n",
       "      <td>197745150</td>\n",
       "      <td>chr1_197743251_197745150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24002</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13328680</td>\n",
       "      <td>13329337</td>\n",
       "      <td>chrY_13328680_13329337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24003</th>\n",
       "      <td>chrY</td>\n",
       "      <td>13370039</td>\n",
       "      <td>13371127</td>\n",
       "      <td>chrY_13370039_13371127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24004</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21153180</td>\n",
       "      <td>21155197</td>\n",
       "      <td>chrY_21153180_21155197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24005</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59213415</td>\n",
       "      <td>59214799</td>\n",
       "      <td>chrY_59213415_59214799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24006</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59362785</td>\n",
       "      <td>59363490</td>\n",
       "      <td>chrY_59362785_59363490</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36068 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr      start        end                        id\n",
       "21750  chr1     858733     861980        chr1_858733_861980\n",
       "21751  chr1    1049470    1052472      chr1_1049470_1052472\n",
       "21752  chr1   10490032   10491779    chr1_10490032_10491779\n",
       "21753  chr1  197169303  197171676  chr1_197169303_197171676\n",
       "21754  chr1  197743251  197745150  chr1_197743251_197745150\n",
       "...     ...        ...        ...                       ...\n",
       "24002  chrY   13328680   13329337    chrY_13328680_13329337\n",
       "24003  chrY   13370039   13371127    chrY_13370039_13371127\n",
       "24004  chrY   21153180   21155197    chrY_21153180_21155197\n",
       "24005  chrY   59213415   59214799    chrY_59213415_59214799\n",
       "24006  chrY   59362785   59363490    chrY_59362785_59363490\n",
       "\n",
       "[36068 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Test elements will be my complete set of active regulatory elements identified in all breast tissues (union)\n",
    "\n",
    "MCF7 = pd.read_table(\"/home/darragh/New/Regulatory_elements/MCF7_reg_elements.txt\", header=None, sep=\"\\t\")\n",
    "T47D = pd.read_table(\"/home/darragh/New/Regulatory_elements/T47D_reg_elements.txt\", header=None, sep=\"\\t\")\n",
    "\n",
    "MCF7.columns = ['chr', 'start', 'end', 'annotation']\n",
    "T47D.columns = ['chr', 'start', 'end', 'annotation']\n",
    "\n",
    "ER_active_promoters = MCF7.loc[(MCF7['annotation'] == \"MCF7_active_promoter\")]\n",
    "ER_active_promoters['annotation'] = \"ER_active_promoter\"\n",
    "\n",
    "ER_polycomb_promoters = MCF7.loc[(MCF7['annotation'] == \"MCF7_polycomb_promoter\")]\n",
    "ER_polycomb_promoters['annotation'] = \"ER_polycomb_promoter\"\n",
    "\n",
    "MCF7_active_enhancers = pybed.BedTool.from_dataframe(MCF7.loc[(MCF7['annotation'] == \"MCF7_active_enhancer\")])\n",
    "\n",
    "T47D_active_enhancers = pybed.BedTool.from_dataframe(T47D.loc[(T47D['annotation'] == \"T47D_active_enhancer\")])\n",
    "\n",
    "ER_active_enhancers = MCF7_active_enhancers.cat(T47D_active_enhancers, postmerge=False)\n",
    "\n",
    "ER_active_enhancers = ER_active_enhancers.sort()\n",
    "\n",
    "ER_active_enhancers = ER_active_enhancers.merge()\n",
    "\n",
    "ER_active_enhancers = pd.read_table(ER_active_enhancers.fn, header=None)\n",
    "\n",
    "ER_active_enhancers.columns = ['chr', 'start', 'end']\n",
    "ER_active_enhancers['annotation'] = \"ER_active_enhancer\"\n",
    "\n",
    "\n",
    "\n",
    "ER_regs = pd.concat([ER_active_promoters, ER_polycomb_promoters, ER_active_enhancers])\n",
    "ER_regs['id'] = ER_regs['chr'] + '_' + ER_regs['start'].astype(str) + '_' + ER_regs['end'].astype(str)\n",
    "ER_regs = ER_regs.drop_duplicates()\n",
    "ER_regs = ER_regs[['chr', 'start', 'end', 'id']]\n",
    "ER_regs.to_csv(\"/home/darragh/New/ActiveDriverWGS/inputs/ER_regs.txt\", sep=\"\\t\", index=False, header=None)\n",
    "display(ER_regs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2606844/1218585893.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  TN_active_promoters['annotation'] = \"TN_active_promoter\"\n",
      "/tmp/ipykernel_2606844/1218585893.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  TN_polycomb_promoters['annotation'] = \"TN_polycomb_promoter\"\n",
      "/usr/lib/python3.8/subprocess.py:853: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n",
      "/usr/lib/python3.8/subprocess.py:853: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25972</th>\n",
       "      <td>chr1</td>\n",
       "      <td>892973</td>\n",
       "      <td>897190</td>\n",
       "      <td>chr1_892973_897190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25973</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1050242</td>\n",
       "      <td>1052504</td>\n",
       "      <td>chr1_1050242_1052504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25974</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10269657</td>\n",
       "      <td>10272253</td>\n",
       "      <td>chr1_10269657_10272253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25975</th>\n",
       "      <td>chr1</td>\n",
       "      <td>160982557</td>\n",
       "      <td>160991041</td>\n",
       "      <td>chr1_160982557_160991041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25976</th>\n",
       "      <td>chr1</td>\n",
       "      <td>160991210</td>\n",
       "      <td>160993142</td>\n",
       "      <td>chr1_160991210_160993142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59113</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59099066</td>\n",
       "      <td>59101389</td>\n",
       "      <td>chrY_59099066_59101389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59114</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59213314</td>\n",
       "      <td>59214942</td>\n",
       "      <td>chrY_59213314_59214942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59115</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59220433</td>\n",
       "      <td>59221668</td>\n",
       "      <td>chrY_59220433_59221668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59116</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59348730</td>\n",
       "      <td>59349716</td>\n",
       "      <td>chrY_59348730_59349716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59117</th>\n",
       "      <td>chrY</td>\n",
       "      <td>59362785</td>\n",
       "      <td>59363494</td>\n",
       "      <td>chrY_59362785_59363494</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71101 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr      start        end                        id\n",
       "25972  chr1     892973     897190        chr1_892973_897190\n",
       "25973  chr1    1050242    1052504      chr1_1050242_1052504\n",
       "25974  chr1   10269657   10272253    chr1_10269657_10272253\n",
       "25975  chr1  160982557  160991041  chr1_160982557_160991041\n",
       "25976  chr1  160991210  160993142  chr1_160991210_160993142\n",
       "...     ...        ...        ...                       ...\n",
       "59113  chrY   59099066   59101389    chrY_59099066_59101389\n",
       "59114  chrY   59213314   59214942    chrY_59213314_59214942\n",
       "59115  chrY   59220433   59221668    chrY_59220433_59221668\n",
       "59116  chrY   59348730   59349716    chrY_59348730_59349716\n",
       "59117  chrY   59362785   59363494    chrY_59362785_59363494\n",
       "\n",
       "[71101 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Test elements will be my complete set of active regulatory elements identified in all breast tissues (union)\n",
    "\n",
    "MDAMB468 = pd.read_table(\"/home/darragh/New/Regulatory_elements/MDAMB468_reg_elements.txt\", header=None, sep=\"\\t\")\n",
    "MDAMB231 = pd.read_table(\"/home/darragh/New/Regulatory_elements/MDAMB231_reg_elements.txt\", header=None, sep=\"\\t\")\n",
    "MDAMB436 = pd.read_table(\"/home/darragh/New/Regulatory_elements/MDAMB436_reg_elements.txt\", header=None, sep=\"\\t\")\n",
    "BT549 = pd.read_table(\"/home/darragh/New/Regulatory_elements/BT549_reg_elements.txt\", header=None, sep=\"\\t\")\n",
    "\n",
    "MDAMB468.columns = ['chr', 'start', 'end', 'annotation']\n",
    "MDAMB231.columns = ['chr', 'start', 'end', 'annotation']\n",
    "MDAMB436.columns = ['chr', 'start', 'end', 'annotation']\n",
    "BT549.columns = ['chr', 'start', 'end', 'annotation']\n",
    "\n",
    "TN_active_promoters = MDAMB468.loc[(MDAMB468['annotation'] == \"MDAMB468_active_promoter\")]\n",
    "TN_active_promoters['annotation'] = \"TN_active_promoter\"\n",
    "\n",
    "TN_polycomb_promoters = MDAMB468.loc[(MDAMB468['annotation'] == \"MDAMB468_polycomb_promoter\")]\n",
    "TN_polycomb_promoters['annotation'] = \"TN_polycomb_promoter\"\n",
    "\n",
    "MDAMB468_active_enhancers = pybed.BedTool.from_dataframe(MDAMB468.loc[(MDAMB468['annotation'] == \"MDAMB468_active_enhancer\")])\n",
    "\n",
    "MDAMB231_active_enhancers = pybed.BedTool.from_dataframe(MDAMB231.loc[(MDAMB231['annotation'] == \"MDAMB231_active_enhancer\")])\n",
    "\n",
    "MDAMB436_active_enhancers = pybed.BedTool.from_dataframe(MDAMB436.loc[(MDAMB436['annotation'] == \"MDAMB436_active_enhancer\")])\n",
    "\n",
    "BT549_active_enhancers = pybed.BedTool.from_dataframe(BT549.loc[(BT549['annotation'] == \"BT549_active_enhancer\")])\n",
    "\n",
    "\n",
    "TN_active_enhancers = MDAMB468_active_enhancers.cat(MDAMB231_active_enhancers, postmerge=False)\n",
    "TN_active_enhancers = TN_active_enhancers.cat(MDAMB436_active_enhancers, postmerge=False)\n",
    "TN_active_enhancers = TN_active_enhancers.cat(BT549_active_enhancers, postmerge=False)\n",
    "\n",
    "TN_active_enhancers = TN_active_enhancers.sort()\n",
    "\n",
    "TN_active_enhancers = TN_active_enhancers.merge()\n",
    "\n",
    "TN_active_enhancers = pd.read_table(TN_active_enhancers.fn, header=None)\n",
    "\n",
    "TN_active_enhancers.columns = ['chr', 'start', 'end']\n",
    "TN_active_enhancers['annotation'] = \"TN_active_enhancer\"\n",
    "\n",
    "\n",
    "TN_regs = pd.concat([TN_active_promoters, TN_polycomb_promoters, TN_active_enhancers])\n",
    "TN_regs['id'] = TN_regs['chr'] + '_' + TN_regs['start'].astype(str) + '_' + TN_regs['end'].astype(str)\n",
    "TN_regs = TN_regs.drop_duplicates()\n",
    "TN_regs = TN_regs[['chr', 'start', 'end', 'id']]\n",
    "TN_regs.to_csv(\"/home/darragh/New/ActiveDriverWGS/inputs/TN_regs.txt\", sep=\"\\t\", index=False, header=None)\n",
    "display(TN_regs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "hg19_exons = pybed.BedTool(\"/index/hg19/hg19_exons.bed\")\n",
    "muts = pybed.BedTool(\"/home/darragh/New/WGS/Hartwig/Hartwig_unfiltered.txt\")\n",
    "\n",
    "muts = muts.subtract(hg19_exons, A=True)\n",
    "muts = pd.read_table(muts.fn, header=None)\n",
    "\n",
    "muts.to_csv(\"/home/darragh/New/WGS/Hartwig/Hartwig_unfiltered_non_coding.txt\", sep=\"\\t\", header=False, index=False)\n",
    "\n",
    "display(muts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2</td>\n",
       "      <td>117070741</td>\n",
       "      <td>117070741</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>WIDE01010341T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr16</td>\n",
       "      <td>49058618</td>\n",
       "      <td>49058618</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>WIDE01010341T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr6</td>\n",
       "      <td>49874973</td>\n",
       "      <td>49874973</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>WIDE01010341T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr10</td>\n",
       "      <td>73674868</td>\n",
       "      <td>73674868</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>WIDE01010341T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr8</td>\n",
       "      <td>88584756</td>\n",
       "      <td>88584756</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>WIDE01010341T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14391922</th>\n",
       "      <td>chr1</td>\n",
       "      <td>219303261</td>\n",
       "      <td>219303261</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02030276T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14391923</th>\n",
       "      <td>chr2</td>\n",
       "      <td>217787763</td>\n",
       "      <td>217787763</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02030276T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14391924</th>\n",
       "      <td>chrX</td>\n",
       "      <td>43606412</td>\n",
       "      <td>43606412</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02030276T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14391925</th>\n",
       "      <td>chr3</td>\n",
       "      <td>166552455</td>\n",
       "      <td>166552455</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CPCT02030276T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14391928</th>\n",
       "      <td>chr8</td>\n",
       "      <td>68658949</td>\n",
       "      <td>68658949</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CPCT02030276T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8459340 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            chr      start        end ref alt        patient\n",
       "1          chr2  117070741  117070741   T   A  WIDE01010341T\n",
       "4         chr16   49058618   49058618   G   A  WIDE01010341T\n",
       "5          chr6   49874973   49874973   C   A  WIDE01010341T\n",
       "6         chr10   73674868   73674868   A   G  WIDE01010341T\n",
       "7          chr8   88584756   88584756   C   T  WIDE01010341T\n",
       "...         ...        ...        ...  ..  ..            ...\n",
       "14391922   chr1  219303261  219303261   G   A  CPCT02030276T\n",
       "14391923   chr2  217787763  217787763   G   A  CPCT02030276T\n",
       "14391924   chrX   43606412   43606412   C   A  CPCT02030276T\n",
       "14391925   chr3  166552455  166552455   G   A  CPCT02030276T\n",
       "14391928   chr8   68658949   68658949   C   T  CPCT02030276T\n",
       "\n",
       "[8459340 rows x 6 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "muts = pd.read_table(\"/home/darragh/New/WGS/Hartwig/Hartwig_unfiltered_non_coding.txt\", header=None)\n",
    "muts.columns = ['chr', 'start', 'end', 'ref', 'alt','patient']\n",
    "\n",
    "muts = muts[muts['ref'].str.len() < 2]\n",
    "muts = muts[muts['alt'].str.len() < 2]\n",
    "muts = muts.loc[muts['ref'] != \"-\"]\n",
    "muts = muts.loc[muts['alt'] != \"-\"]\n",
    "muts['patient'] = muts['patient'].str.split('.').str[0]\n",
    "\n",
    "#Hypermutation cut off of 20mt per Mb\n",
    "#Hypermutated\n",
    "values_to_remove = [\"CPCT02010520T\", \"CPCT02010731T\", \"CPCT02010896T\", \"CPCT02020215T\", \"CPCT02020299T\", \"CPCT02020344T\",\n",
    "                    \"CPCT02020436T\", \"CPCT02020514T\", \"CPCT02020680T\", \"CPCT02030242T\", \"CPCT02030265TII\",\n",
    "                    \"CPCT02030521T\", \"CPCT02050299T\", \"CPCT02050383T\", \"CPCT02080056T\", \"CPCT02080153T\",\n",
    "                    \"CPCT02120108T\", \"CPCT02210004T\", \"CPCT02220029T\", \"CPCT02310004T\", \"CPCT02330076T\",\n",
    "                    \"CPCT02350008T\", \"CPCT02380035T\", \"CPCT02390015T\", \"DRUP01010037T\", \"DRUP01010148T\",\n",
    "                    \"DRUP01060008T\", \"DRUP01080004T\", \"DRUP01110007T\", \"DRUP01220001T\", \"DRUP01220003T\",\n",
    "                    \"DRUP01330020T\", \"WIDE01010069T\"]\n",
    "\n",
    "# Remove rows based on values in the \"patient\" column\n",
    "muts = muts[~muts[\"patient\"].isin(values_to_remove)]\n",
    "\n",
    "#Duplicated\n",
    "values_to_remove = [\"CPCT02120047TII\",\"CPCT02020186TII\", \"CPCT02340066TII\", \"CPCT02340066TIII\",\"CPCT02340066TIV\", \"CPCT02030265TII\", \"CPCT02030218TIII\", \"CPCT02030289TII\", \"CPCT02030289TIII\", \"CPCT02020345TII\",\n",
    "  \"CPCT02080029TII\", \"CPCT02080014TII\" , \"CPCT02080005TII\", \"CPCT02080001TII\", \"CPCT02080070TII\", \"CPCT02100001TIII\", \"CPCT02100067TII\", \"CPCT02020245TII\", \"CPCT02020245TIII\", \"CPCT02140004TII\", \"CPCT02140085TII\", \"CPCT02140091TII\",\n",
    "  \"CPCT02010359TII\", \"CPCT02020597TII\", \"CPCT02020597TIII\", \"CPCT02010337TII\", \"CPCT02040071TII\", \"CPCT02010267TIII\", \"CPCT02390001TII\", \"CPCT02020493TII\", \"CPCT02230120TII\", \"CPCT02010276TII\", \"CPCT02010288TII\"]\n",
    "\n",
    "# Remove rows based on values in the \"patient\" column\n",
    "muts = muts[~muts[\"patient\"].isin(values_to_remove)]\n",
    "\n",
    "muts.to_csv(\"/home/darragh/New/WGS/Hartwig/Hartwig_unfiltered_non_coding_non_hyper_non_dup.txt\", sep=\"\\t\", header=False, index=False)\n",
    "\n",
    "display(muts)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>pos1</th><th scope=col>pos2</th><th scope=col>ref</th><th scope=col>alt</th><th scope=col>patient</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr2 </td><td>117070741</td><td>117070741</td><td>T</td><td>A</td><td>WIDE01010341T</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr16</td><td> 49058618</td><td> 49058618</td><td>G</td><td>A</td><td>WIDE01010341T</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr6 </td><td> 49874973</td><td> 49874973</td><td>C</td><td>A</td><td>WIDE01010341T</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr10</td><td> 73674868</td><td> 73674868</td><td>A</td><td>G</td><td>WIDE01010341T</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr8 </td><td> 88584756</td><td> 88584756</td><td>C</td><td>T</td><td>WIDE01010341T</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr14</td><td> 80075379</td><td> 80075379</td><td>A</td><td>T</td><td>WIDE01010341T</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & chr & pos1 & pos2 & ref & alt & patient\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr2  & 117070741 & 117070741 & T & A & WIDE01010341T\\\\\n",
       "\t2 & chr16 &  49058618 &  49058618 & G & A & WIDE01010341T\\\\\n",
       "\t3 & chr6  &  49874973 &  49874973 & C & A & WIDE01010341T\\\\\n",
       "\t4 & chr10 &  73674868 &  73674868 & A & G & WIDE01010341T\\\\\n",
       "\t5 & chr8  &  88584756 &  88584756 & C & T & WIDE01010341T\\\\\n",
       "\t6 & chr14 &  80075379 &  80075379 & A & T & WIDE01010341T\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | chr &lt;chr&gt; | pos1 &lt;int&gt; | pos2 &lt;int&gt; | ref &lt;chr&gt; | alt &lt;chr&gt; | patient &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | chr2  | 117070741 | 117070741 | T | A | WIDE01010341T |\n",
       "| 2 | chr16 |  49058618 |  49058618 | G | A | WIDE01010341T |\n",
       "| 3 | chr6  |  49874973 |  49874973 | C | A | WIDE01010341T |\n",
       "| 4 | chr10 |  73674868 |  73674868 | A | G | WIDE01010341T |\n",
       "| 5 | chr8  |  88584756 |  88584756 | C | T | WIDE01010341T |\n",
       "| 6 | chr14 |  80075379 |  80075379 | A | T | WIDE01010341T |\n",
       "\n"
      ],
      "text/plain": [
       "  chr   pos1      pos2      ref alt patient      \n",
       "1 chr2  117070741 117070741 T   A   WIDE01010341T\n",
       "2 chr16  49058618  49058618 G   A   WIDE01010341T\n",
       "3 chr6   49874973  49874973 C   A   WIDE01010341T\n",
       "4 chr10  73674868  73674868 A   G   WIDE01010341T\n",
       "5 chr8   88584756  88584756 C   T   WIDE01010341T\n",
       "6 chr14  80075379  80075379 A   T   WIDE01010341T"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“multiple methods tables found for ‘aperm’”\n",
      "Warning message:\n",
      "“replacing previous import ‘BiocGenerics::aperm’ by ‘DelayedArray::aperm’ when loading ‘SummarizedExperiment’”\n",
      "Warning message:\n",
      "“no function found corresponding to methods exports from ‘BSgenome’ for: ‘releaseName’”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 remove hypermut, n= 0 ,  0 %\n",
      "hypermuted samples:   \n",
      "\n",
      "reversing 0 positions\n",
      "Removing  0  invalid SNVs & indels\n",
      "\n",
      "Number of Elements with 0 Mutations:  9494 \n",
      "Tests to do:  89580 \n"
     ]
    }
   ],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "#Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/WGS/Hartwig/Hartwig_unfiltered_non_coding_non_hyper.txt\",header = FALSE, sep = \"\\t\")\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/WGS/Hartwig/Hartwig_unfiltered_non_coding_non_hyper_non_dup.txt\",header = FALSE, sep = \"\\t\")\n",
    "#Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/WGS/ICGC/daragh_icgc_brca_cohort_mutations_icgc_3.non_coding_non_hyper.tsv\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "head(Hartwig_all_non_coding)\n",
    "\n",
    "# #Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/All_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "\n",
    "Primary_regs = Primary_regs[,c(\"chr\", \"start\", \"end\", \"id\")]\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 42,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "#write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/All_regs_hartwig_non_hyper.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)\n",
    "#write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/All_regs_ICGC_non_hyper.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/All_regs_hartwig_non_hyper_non_dup.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65419</td>\n",
       "      <td>71585</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450703</td>\n",
       "      <td>451697</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>685679</td>\n",
       "      <td>686673</td>\n",
       "      <td>OR4F16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>923928</td>\n",
       "      <td>944581</td>\n",
       "      <td>SAMD11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>944203</td>\n",
       "      <td>959309</td>\n",
       "      <td>NOC2L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19981</th>\n",
       "      <td>chrKI270726.1</td>\n",
       "      <td>26241</td>\n",
       "      <td>26534</td>\n",
       "      <td>AC233755.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19982</th>\n",
       "      <td>chrKI270726.1</td>\n",
       "      <td>41444</td>\n",
       "      <td>41876</td>\n",
       "      <td>AC233755.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19983</th>\n",
       "      <td>chrKI270711.1</td>\n",
       "      <td>4612</td>\n",
       "      <td>29626</td>\n",
       "      <td>AC240274.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19984</th>\n",
       "      <td>chrKI270713.1</td>\n",
       "      <td>31698</td>\n",
       "      <td>32528</td>\n",
       "      <td>AC213203.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19985</th>\n",
       "      <td>chrKI270713.1</td>\n",
       "      <td>35407</td>\n",
       "      <td>35916</td>\n",
       "      <td>AC213203.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19986 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 chr   start     end        gene\n",
       "0               chr1   65419   71585       OR4F5\n",
       "1               chr1  450703  451697      OR4F29\n",
       "2               chr1  685679  686673      OR4F16\n",
       "3               chr1  923928  944581      SAMD11\n",
       "4               chr1  944203  959309       NOC2L\n",
       "...              ...     ...     ...         ...\n",
       "19981  chrKI270726.1   26241   26534  AC233755.2\n",
       "19982  chrKI270726.1   41444   41876  AC233755.1\n",
       "19983  chrKI270711.1    4612   29626  AC240274.1\n",
       "19984  chrKI270713.1   31698   32528  AC213203.2\n",
       "19985  chrKI270713.1   35407   35916  AC213203.1\n",
       "\n",
       "[19986 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65419</td>\n",
       "      <td>65433</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>65520</td>\n",
       "      <td>65573</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69037</td>\n",
       "      <td>71585</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69055</td>\n",
       "      <td>70108</td>\n",
       "      <td>OR4F5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>450703</td>\n",
       "      <td>451697</td>\n",
       "      <td>OR4F29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136146</th>\n",
       "      <td>chrKI270711.1</td>\n",
       "      <td>7817</td>\n",
       "      <td>7930</td>\n",
       "      <td>AC240274.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136147</th>\n",
       "      <td>chrKI270711.1</td>\n",
       "      <td>6991</td>\n",
       "      <td>7102</td>\n",
       "      <td>AC240274.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136148</th>\n",
       "      <td>chrKI270711.1</td>\n",
       "      <td>6102</td>\n",
       "      <td>6370</td>\n",
       "      <td>AC240274.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136149</th>\n",
       "      <td>chrKI270713.1</td>\n",
       "      <td>31698</td>\n",
       "      <td>32528</td>\n",
       "      <td>AC213203.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136150</th>\n",
       "      <td>chrKI270713.1</td>\n",
       "      <td>35407</td>\n",
       "      <td>35916</td>\n",
       "      <td>AC213203.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1136151 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   chr   start     end        gene\n",
       "0                 chr1   65419   65433       OR4F5\n",
       "1                 chr1   65520   65573       OR4F5\n",
       "2                 chr1   69037   71585       OR4F5\n",
       "3                 chr1   69055   70108       OR4F5\n",
       "4                 chr1  450703  451697      OR4F29\n",
       "...                ...     ...     ...         ...\n",
       "1136146  chrKI270711.1    7817    7930  AC240274.1\n",
       "1136147  chrKI270711.1    6991    7102  AC240274.1\n",
       "1136148  chrKI270711.1    6102    6370  AC240274.1\n",
       "1136149  chrKI270713.1   31698   32528  AC213203.2\n",
       "1136150  chrKI270713.1   35407   35916  AC213203.1\n",
       "\n",
       "[1136151 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "hg38_genes = pd.read_table(\"/home/darragh/New/OnkDriveFML/PDAC_maeve/hg38_genes\", sep=\"\\t\", header=None)\n",
    "hg38_genes.columns = ['chr', 'start', 'end', 'gene']\n",
    "hg38_genes['gene'] = hg38_genes['gene'].str.split(\";\", expand=True)[0]\n",
    "hg38_genes.to_csv(\"/home/darragh/New/OnkDriveFML/PDAC_maeve/hg38_genes.txt\", sep=\"\\t\", header=False, index=False)\n",
    "\n",
    "hg38_exons = pd.read_table(\"/home/darragh/New/OnkDriveFML/PDAC_maeve/hg38_exons\", sep=\"\\t\", header=None)\n",
    "hg38_exons.columns = ['chr', 'start', 'end', 'gene']\n",
    "hg38_exons['gene'] = hg38_exons['gene'].str.split(\";\", expand=True)[0]\n",
    "hg38_exons.to_csv(\"/home/darragh/New/OnkDriveFML/PDAC_maeve/hg38_exons.txt\", sep=\"\\t\", header=False, index=False)\n",
    "\n",
    "display(hg38_genes)\n",
    "display(hg38_exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 remove hypermut, n= 0 ,  0 %\n",
      "hypermuted samples:   \n",
      "\n",
      "reversing 0 positions\n",
      "Removing  0  invalid SNVs & indels\n",
      "\n",
      "Number of Elements with 0 Mutations:  519 \n",
      "Tests to do:  19410 \n"
     ]
    }
   ],
   "source": [
    "#Mutations\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "muts = read.table(file=\"/home/darragh/New/PDAC_WGS/VCFs/ActiveDriver_muts.csv\",header = FALSE, sep = \"\\t\")\n",
    "colnames(muts) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "muts = muts[muts$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "muts$pos1 = as.integer(muts$pos1)\n",
    "muts$pos2 = as.integer(muts$pos2)\n",
    "\n",
    "#Regions\n",
    "Regions = read.table(file=\"/home/darragh/New/OnkDriveFML/PDAC_maeve/hg38_genes.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Regions) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Regions = Regions[Regions$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "#Sites\n",
    "Sites = read.table(file=\"/home/darragh/New/OnkDriveFML/PDAC_maeve/hg38_exons.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Sites) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Sites = Sites[Sites$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "Sites = unique(Sites)\n",
    "\n",
    "results = ActiveDriverWGS(mutations = muts,\n",
    "                          elements = Regions,\n",
    "                        \n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 40,\n",
    "                          ref_genome = \"hg38\")\n",
    "\n",
    "#write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/ER_regs_ICGC.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 100 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>id</th><th scope=col>pp_element</th><th scope=col>element_muts_obs</th><th scope=col>element_muts_exp</th><th scope=col>element_enriched</th><th scope=col>pp_site</th><th scope=col>site_muts_obs</th><th scope=col>site_muts_exp</th><th scope=col>site_enriched</th><th scope=col>fdr_element</th><th scope=col>fdr_site</th><th scope=col>has_site_mutations</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>PABPC1L2A</td><td>1.396447e-74</td><td>45</td><td> 1.0346828</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.782979e-70</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>2</th><td>CT47B1   </td><td>5.499257e-40</td><td>35</td><td> 3.0487297</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>5.479734e-36</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>3</th><td>PRR32    </td><td>3.949606e-39</td><td>49</td><td> 3.9203678</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.623723e-35</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>4</th><td>FAM47C   </td><td>7.746220e-39</td><td>50</td><td> 5.3386166</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>3.859361e-35</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>5</th><td>TSPYL5   </td><td>3.546338e-38</td><td>57</td><td> 6.8420465</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.413499e-34</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>6</th><td>FABP5    </td><td>8.191851e-38</td><td>58</td><td> 7.7089385</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.720923e-34</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>7</th><td>OR4A47   </td><td>1.238507e-37</td><td>38</td><td> 1.9170070</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>3.526031e-34</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>8</th><td>PABPC4L  </td><td>9.526415e-36</td><td>57</td><td> 7.2895514</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.373149e-32</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>9</th><td>GYPB     </td><td>1.191023e-35</td><td>55</td><td>11.1995994</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.637323e-32</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>10</th><td>DIO3     </td><td>1.932920e-35</td><td>44</td><td> 3.9130694</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>3.852115e-32</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>11</th><td>TMEM215  </td><td>2.577214e-35</td><td>57</td><td> 7.5837952</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>4.669209e-32</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>12</th><td>REG3G    </td><td>6.070298e-35</td><td>51</td><td> 5.4927299</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.008125e-31</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>13</th><td>GYPA     </td><td>8.332016e-34</td><td>58</td><td>14.2045127</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.277298e-30</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>14</th><td>RTL3     </td><td>1.225487e-33</td><td>49</td><td> 5.3182638</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.744480e-30</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>15</th><td>C10orf142</td><td>8.768327e-33</td><td>50</td><td> 6.2485105</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.164960e-29</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>16</th><td>ZNF66    </td><td>1.472389e-32</td><td>58</td><td>14.9259388</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.833952e-29</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>17</th><td>CXCR4    </td><td>1.716483e-32</td><td>49</td><td> 7.4614960</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.012223e-29</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>18</th><td>PABPC1L2B</td><td>1.848870e-32</td><td>23</td><td> 0.9542834</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.047008e-29</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>19</th><td>MTRNR2L13</td><td>2.127050e-32</td><td>58</td><td> 7.7674464</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.231051e-29</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>20</th><td>MARCKS   </td><td>2.704414e-32</td><td>58</td><td>10.8127837</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.694814e-29</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>21</th><td>TIPARP   </td><td>1.034107e-31</td><td>58</td><td>16.4674615</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>9.813674e-29</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>22</th><td>AGTR2    </td><td>2.877050e-31</td><td>58</td><td> 8.5752236</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.606215e-28</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>23</th><td>MTRNR2L6 </td><td>3.881015e-30</td><td>22</td><td> 0.6489617</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>3.362815e-27</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>24</th><td>FOXR2    </td><td>8.128059e-30</td><td>38</td><td> 3.1681834</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>6.749337e-27</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>25</th><td>NPBWR1   </td><td>8.955406e-30</td><td>58</td><td> 9.3804133</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>7.138892e-27</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>26</th><td>BTG1     </td><td>1.079333e-29</td><td>57</td><td> 9.7249912</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>8.273084e-27</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>27</th><td>MAGEC2   </td><td>5.106952e-29</td><td>49</td><td> 6.5372291</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>3.769498e-26</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>28</th><td>MTRNR2L12</td><td>5.333579e-29</td><td>49</td><td> 6.0658844</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>3.796175e-26</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>29</th><td>SLITRK3  </td><td>7.334380e-29</td><td>58</td><td>12.3594610</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>5.040236e-26</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>30</th><td>CEBPA    </td><td>8.781243e-29</td><td>52</td><td> 8.6121095</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>5.833380e-26</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>71</th><td>SDHAF3  </td><td>8.052481e-22</td><td>58</td><td>23.747352</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.280830e-19</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>72</th><td>CSTF2T  </td><td>9.197873e-22</td><td>40</td><td> 6.198350</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.545895e-19</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>73</th><td>PTGS2   </td><td>9.921123e-22</td><td>58</td><td>14.252650</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.708467e-19</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>74</th><td>GPX6    </td><td>1.040557e-21</td><td>52</td><td>14.488189</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.802333e-19</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>75</th><td>KCNA5   </td><td>1.057900e-21</td><td>49</td><td> 9.525230</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.811051e-19</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>76</th><td>ZXDB    </td><td>2.205257e-21</td><td>40</td><td> 7.520052</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>5.782705e-19</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>77</th><td>FOXB2   </td><td>2.351991e-21</td><td>33</td><td> 4.287914</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>6.087380e-19</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>78</th><td>HDGFL1  </td><td>2.531268e-21</td><td>46</td><td> 9.643443</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>6.467390e-19</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>79</th><td>NEUROD6 </td><td>7.452465e-21</td><td>47</td><td> 8.717012</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.880002e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>80</th><td>RPRM    </td><td>7.885459e-21</td><td>36</td><td> 5.568092</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.964366e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>81</th><td>SOX1    </td><td>1.056835e-20</td><td>42</td><td> 7.776263</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.600204e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>82</th><td>TAS2R16 </td><td>1.463009e-20</td><td>22</td><td> 1.332422</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>3.555646e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>83</th><td>DUX4    </td><td>1.691051e-20</td><td>57</td><td>24.840549</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>4.060355e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>84</th><td>PCDH10  </td><td>1.758555e-20</td><td>58</td><td>23.111651</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>4.172172e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>85</th><td>VCX2    </td><td>1.783939e-20</td><td>38</td><td> 5.687692</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>4.182602e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>86</th><td>RAB6C   </td><td>2.096076e-20</td><td>55</td><td>12.277323</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>4.857289e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>87</th><td>SLC6A15 </td><td>2.425812e-20</td><td>58</td><td>22.830201</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>5.556783e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>88</th><td>HAND2   </td><td>2.653981e-20</td><td>57</td><td>14.147082</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>5.942831e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>89</th><td>SERPINA7</td><td>2.632022e-20</td><td>44</td><td> 8.436398</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>5.942831e-18</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>90</th><td>HSPA13  </td><td>5.288513e-20</td><td>58</td><td>15.913189</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.171053e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>91</th><td>TMEM229A</td><td>5.544651e-20</td><td>48</td><td>10.625976</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.214279e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>92</th><td>OR4C5   </td><td>6.026160e-20</td><td>34</td><td> 4.290414</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.305384e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>93</th><td>DDIT4L  </td><td>6.229151e-20</td><td>56</td><td>13.453267</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.334847e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>94</th><td>ZBTB18  </td><td>8.589195e-20</td><td>55</td><td>14.259962</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>1.821001e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>95</th><td>OR4C13  </td><td>9.601452e-20</td><td>31</td><td> 3.509284</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.014182e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>96</th><td>ITM2A   </td><td>1.137988e-19</td><td>54</td><td>13.179017</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.362391e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>97</th><td>AVPR1A  </td><td>1.279005e-19</td><td>56</td><td>15.392623</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>2.627763e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>98</th><td>CAVIN4  </td><td>1.582745e-19</td><td>55</td><td>13.976276</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>3.218626e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>99</th><td>ADRA2C  </td><td>3.445722e-19</td><td>52</td><td>12.225444</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>6.936342e-17</td><td>1</td><td></td></tr>\n",
       "\t<tr><th scope=row>100</th><td>TCF21   </td><td>3.793826e-19</td><td>58</td><td>15.381852</td><td>TRUE</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>7.560716e-17</td><td>1</td><td></td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 100 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & id & pp\\_element & element\\_muts\\_obs & element\\_muts\\_exp & element\\_enriched & pp\\_site & site\\_muts\\_obs & site\\_muts\\_exp & site\\_enriched & fdr\\_element & fdr\\_site & has\\_site\\_mutations\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl> & <lgl> & <dbl> & <dbl> & <dbl> & <lgl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & PABPC1L2A & 1.396447e-74 & 45 &  1.0346828 & TRUE & 1 & NA & NA & NA & 2.782979e-70 & 1 & \\\\\n",
       "\t2 & CT47B1    & 5.499257e-40 & 35 &  3.0487297 & TRUE & 1 & NA & NA & NA & 5.479734e-36 & 1 & \\\\\n",
       "\t3 & PRR32     & 3.949606e-39 & 49 &  3.9203678 & TRUE & 1 & NA & NA & NA & 2.623723e-35 & 1 & \\\\\n",
       "\t4 & FAM47C    & 7.746220e-39 & 50 &  5.3386166 & TRUE & 1 & NA & NA & NA & 3.859361e-35 & 1 & \\\\\n",
       "\t5 & TSPYL5    & 3.546338e-38 & 57 &  6.8420465 & TRUE & 1 & NA & NA & NA & 1.413499e-34 & 1 & \\\\\n",
       "\t6 & FABP5     & 8.191851e-38 & 58 &  7.7089385 & TRUE & 1 & NA & NA & NA & 2.720923e-34 & 1 & \\\\\n",
       "\t7 & OR4A47    & 1.238507e-37 & 38 &  1.9170070 & TRUE & 1 & NA & NA & NA & 3.526031e-34 & 1 & \\\\\n",
       "\t8 & PABPC4L   & 9.526415e-36 & 57 &  7.2895514 & TRUE & 1 & NA & NA & NA & 2.373149e-32 & 1 & \\\\\n",
       "\t9 & GYPB      & 1.191023e-35 & 55 & 11.1995994 & TRUE & 1 & NA & NA & NA & 2.637323e-32 & 1 & \\\\\n",
       "\t10 & DIO3      & 1.932920e-35 & 44 &  3.9130694 & TRUE & 1 & NA & NA & NA & 3.852115e-32 & 1 & \\\\\n",
       "\t11 & TMEM215   & 2.577214e-35 & 57 &  7.5837952 & TRUE & 1 & NA & NA & NA & 4.669209e-32 & 1 & \\\\\n",
       "\t12 & REG3G     & 6.070298e-35 & 51 &  5.4927299 & TRUE & 1 & NA & NA & NA & 1.008125e-31 & 1 & \\\\\n",
       "\t13 & GYPA      & 8.332016e-34 & 58 & 14.2045127 & TRUE & 1 & NA & NA & NA & 1.277298e-30 & 1 & \\\\\n",
       "\t14 & RTL3      & 1.225487e-33 & 49 &  5.3182638 & TRUE & 1 & NA & NA & NA & 1.744480e-30 & 1 & \\\\\n",
       "\t15 & C10orf142 & 8.768327e-33 & 50 &  6.2485105 & TRUE & 1 & NA & NA & NA & 1.164960e-29 & 1 & \\\\\n",
       "\t16 & ZNF66     & 1.472389e-32 & 58 & 14.9259388 & TRUE & 1 & NA & NA & NA & 1.833952e-29 & 1 & \\\\\n",
       "\t17 & CXCR4     & 1.716483e-32 & 49 &  7.4614960 & TRUE & 1 & NA & NA & NA & 2.012223e-29 & 1 & \\\\\n",
       "\t18 & PABPC1L2B & 1.848870e-32 & 23 &  0.9542834 & TRUE & 1 & NA & NA & NA & 2.047008e-29 & 1 & \\\\\n",
       "\t19 & MTRNR2L13 & 2.127050e-32 & 58 &  7.7674464 & TRUE & 1 & NA & NA & NA & 2.231051e-29 & 1 & \\\\\n",
       "\t20 & MARCKS    & 2.704414e-32 & 58 & 10.8127837 & TRUE & 1 & NA & NA & NA & 2.694814e-29 & 1 & \\\\\n",
       "\t21 & TIPARP    & 1.034107e-31 & 58 & 16.4674615 & TRUE & 1 & NA & NA & NA & 9.813674e-29 & 1 & \\\\\n",
       "\t22 & AGTR2     & 2.877050e-31 & 58 &  8.5752236 & TRUE & 1 & NA & NA & NA & 2.606215e-28 & 1 & \\\\\n",
       "\t23 & MTRNR2L6  & 3.881015e-30 & 22 &  0.6489617 & TRUE & 1 & NA & NA & NA & 3.362815e-27 & 1 & \\\\\n",
       "\t24 & FOXR2     & 8.128059e-30 & 38 &  3.1681834 & TRUE & 1 & NA & NA & NA & 6.749337e-27 & 1 & \\\\\n",
       "\t25 & NPBWR1    & 8.955406e-30 & 58 &  9.3804133 & TRUE & 1 & NA & NA & NA & 7.138892e-27 & 1 & \\\\\n",
       "\t26 & BTG1      & 1.079333e-29 & 57 &  9.7249912 & TRUE & 1 & NA & NA & NA & 8.273084e-27 & 1 & \\\\\n",
       "\t27 & MAGEC2    & 5.106952e-29 & 49 &  6.5372291 & TRUE & 1 & NA & NA & NA & 3.769498e-26 & 1 & \\\\\n",
       "\t28 & MTRNR2L12 & 5.333579e-29 & 49 &  6.0658844 & TRUE & 1 & NA & NA & NA & 3.796175e-26 & 1 & \\\\\n",
       "\t29 & SLITRK3   & 7.334380e-29 & 58 & 12.3594610 & TRUE & 1 & NA & NA & NA & 5.040236e-26 & 1 & \\\\\n",
       "\t30 & CEBPA     & 8.781243e-29 & 52 &  8.6121095 & TRUE & 1 & NA & NA & NA & 5.833380e-26 & 1 & \\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t71 & SDHAF3   & 8.052481e-22 & 58 & 23.747352 & TRUE & 1 & NA & NA & NA & 2.280830e-19 & 1 & \\\\\n",
       "\t72 & CSTF2T   & 9.197873e-22 & 40 &  6.198350 & TRUE & 1 & NA & NA & NA & 2.545895e-19 & 1 & \\\\\n",
       "\t73 & PTGS2    & 9.921123e-22 & 58 & 14.252650 & TRUE & 1 & NA & NA & NA & 2.708467e-19 & 1 & \\\\\n",
       "\t74 & GPX6     & 1.040557e-21 & 52 & 14.488189 & TRUE & 1 & NA & NA & NA & 2.802333e-19 & 1 & \\\\\n",
       "\t75 & KCNA5    & 1.057900e-21 & 49 &  9.525230 & TRUE & 1 & NA & NA & NA & 2.811051e-19 & 1 & \\\\\n",
       "\t76 & ZXDB     & 2.205257e-21 & 40 &  7.520052 & TRUE & 1 & NA & NA & NA & 5.782705e-19 & 1 & \\\\\n",
       "\t77 & FOXB2    & 2.351991e-21 & 33 &  4.287914 & TRUE & 1 & NA & NA & NA & 6.087380e-19 & 1 & \\\\\n",
       "\t78 & HDGFL1   & 2.531268e-21 & 46 &  9.643443 & TRUE & 1 & NA & NA & NA & 6.467390e-19 & 1 & \\\\\n",
       "\t79 & NEUROD6  & 7.452465e-21 & 47 &  8.717012 & TRUE & 1 & NA & NA & NA & 1.880002e-18 & 1 & \\\\\n",
       "\t80 & RPRM     & 7.885459e-21 & 36 &  5.568092 & TRUE & 1 & NA & NA & NA & 1.964366e-18 & 1 & \\\\\n",
       "\t81 & SOX1     & 1.056835e-20 & 42 &  7.776263 & TRUE & 1 & NA & NA & NA & 2.600204e-18 & 1 & \\\\\n",
       "\t82 & TAS2R16  & 1.463009e-20 & 22 &  1.332422 & TRUE & 1 & NA & NA & NA & 3.555646e-18 & 1 & \\\\\n",
       "\t83 & DUX4     & 1.691051e-20 & 57 & 24.840549 & TRUE & 1 & NA & NA & NA & 4.060355e-18 & 1 & \\\\\n",
       "\t84 & PCDH10   & 1.758555e-20 & 58 & 23.111651 & TRUE & 1 & NA & NA & NA & 4.172172e-18 & 1 & \\\\\n",
       "\t85 & VCX2     & 1.783939e-20 & 38 &  5.687692 & TRUE & 1 & NA & NA & NA & 4.182602e-18 & 1 & \\\\\n",
       "\t86 & RAB6C    & 2.096076e-20 & 55 & 12.277323 & TRUE & 1 & NA & NA & NA & 4.857289e-18 & 1 & \\\\\n",
       "\t87 & SLC6A15  & 2.425812e-20 & 58 & 22.830201 & TRUE & 1 & NA & NA & NA & 5.556783e-18 & 1 & \\\\\n",
       "\t88 & HAND2    & 2.653981e-20 & 57 & 14.147082 & TRUE & 1 & NA & NA & NA & 5.942831e-18 & 1 & \\\\\n",
       "\t89 & SERPINA7 & 2.632022e-20 & 44 &  8.436398 & TRUE & 1 & NA & NA & NA & 5.942831e-18 & 1 & \\\\\n",
       "\t90 & HSPA13   & 5.288513e-20 & 58 & 15.913189 & TRUE & 1 & NA & NA & NA & 1.171053e-17 & 1 & \\\\\n",
       "\t91 & TMEM229A & 5.544651e-20 & 48 & 10.625976 & TRUE & 1 & NA & NA & NA & 1.214279e-17 & 1 & \\\\\n",
       "\t92 & OR4C5    & 6.026160e-20 & 34 &  4.290414 & TRUE & 1 & NA & NA & NA & 1.305384e-17 & 1 & \\\\\n",
       "\t93 & DDIT4L   & 6.229151e-20 & 56 & 13.453267 & TRUE & 1 & NA & NA & NA & 1.334847e-17 & 1 & \\\\\n",
       "\t94 & ZBTB18   & 8.589195e-20 & 55 & 14.259962 & TRUE & 1 & NA & NA & NA & 1.821001e-17 & 1 & \\\\\n",
       "\t95 & OR4C13   & 9.601452e-20 & 31 &  3.509284 & TRUE & 1 & NA & NA & NA & 2.014182e-17 & 1 & \\\\\n",
       "\t96 & ITM2A    & 1.137988e-19 & 54 & 13.179017 & TRUE & 1 & NA & NA & NA & 2.362391e-17 & 1 & \\\\\n",
       "\t97 & AVPR1A   & 1.279005e-19 & 56 & 15.392623 & TRUE & 1 & NA & NA & NA & 2.627763e-17 & 1 & \\\\\n",
       "\t98 & CAVIN4   & 1.582745e-19 & 55 & 13.976276 & TRUE & 1 & NA & NA & NA & 3.218626e-17 & 1 & \\\\\n",
       "\t99 & ADRA2C   & 3.445722e-19 & 52 & 12.225444 & TRUE & 1 & NA & NA & NA & 6.936342e-17 & 1 & \\\\\n",
       "\t100 & TCF21    & 3.793826e-19 & 58 & 15.381852 & TRUE & 1 & NA & NA & NA & 7.560716e-17 & 1 & \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 100 × 12\n",
       "\n",
       "| <!--/--> | id &lt;chr&gt; | pp_element &lt;dbl&gt; | element_muts_obs &lt;dbl&gt; | element_muts_exp &lt;dbl&gt; | element_enriched &lt;lgl&gt; | pp_site &lt;dbl&gt; | site_muts_obs &lt;dbl&gt; | site_muts_exp &lt;dbl&gt; | site_enriched &lt;lgl&gt; | fdr_element &lt;dbl&gt; | fdr_site &lt;dbl&gt; | has_site_mutations &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | PABPC1L2A | 1.396447e-74 | 45 |  1.0346828 | TRUE | 1 | NA | NA | NA | 2.782979e-70 | 1 | <!----> |\n",
       "| 2 | CT47B1    | 5.499257e-40 | 35 |  3.0487297 | TRUE | 1 | NA | NA | NA | 5.479734e-36 | 1 | <!----> |\n",
       "| 3 | PRR32     | 3.949606e-39 | 49 |  3.9203678 | TRUE | 1 | NA | NA | NA | 2.623723e-35 | 1 | <!----> |\n",
       "| 4 | FAM47C    | 7.746220e-39 | 50 |  5.3386166 | TRUE | 1 | NA | NA | NA | 3.859361e-35 | 1 | <!----> |\n",
       "| 5 | TSPYL5    | 3.546338e-38 | 57 |  6.8420465 | TRUE | 1 | NA | NA | NA | 1.413499e-34 | 1 | <!----> |\n",
       "| 6 | FABP5     | 8.191851e-38 | 58 |  7.7089385 | TRUE | 1 | NA | NA | NA | 2.720923e-34 | 1 | <!----> |\n",
       "| 7 | OR4A47    | 1.238507e-37 | 38 |  1.9170070 | TRUE | 1 | NA | NA | NA | 3.526031e-34 | 1 | <!----> |\n",
       "| 8 | PABPC4L   | 9.526415e-36 | 57 |  7.2895514 | TRUE | 1 | NA | NA | NA | 2.373149e-32 | 1 | <!----> |\n",
       "| 9 | GYPB      | 1.191023e-35 | 55 | 11.1995994 | TRUE | 1 | NA | NA | NA | 2.637323e-32 | 1 | <!----> |\n",
       "| 10 | DIO3      | 1.932920e-35 | 44 |  3.9130694 | TRUE | 1 | NA | NA | NA | 3.852115e-32 | 1 | <!----> |\n",
       "| 11 | TMEM215   | 2.577214e-35 | 57 |  7.5837952 | TRUE | 1 | NA | NA | NA | 4.669209e-32 | 1 | <!----> |\n",
       "| 12 | REG3G     | 6.070298e-35 | 51 |  5.4927299 | TRUE | 1 | NA | NA | NA | 1.008125e-31 | 1 | <!----> |\n",
       "| 13 | GYPA      | 8.332016e-34 | 58 | 14.2045127 | TRUE | 1 | NA | NA | NA | 1.277298e-30 | 1 | <!----> |\n",
       "| 14 | RTL3      | 1.225487e-33 | 49 |  5.3182638 | TRUE | 1 | NA | NA | NA | 1.744480e-30 | 1 | <!----> |\n",
       "| 15 | C10orf142 | 8.768327e-33 | 50 |  6.2485105 | TRUE | 1 | NA | NA | NA | 1.164960e-29 | 1 | <!----> |\n",
       "| 16 | ZNF66     | 1.472389e-32 | 58 | 14.9259388 | TRUE | 1 | NA | NA | NA | 1.833952e-29 | 1 | <!----> |\n",
       "| 17 | CXCR4     | 1.716483e-32 | 49 |  7.4614960 | TRUE | 1 | NA | NA | NA | 2.012223e-29 | 1 | <!----> |\n",
       "| 18 | PABPC1L2B | 1.848870e-32 | 23 |  0.9542834 | TRUE | 1 | NA | NA | NA | 2.047008e-29 | 1 | <!----> |\n",
       "| 19 | MTRNR2L13 | 2.127050e-32 | 58 |  7.7674464 | TRUE | 1 | NA | NA | NA | 2.231051e-29 | 1 | <!----> |\n",
       "| 20 | MARCKS    | 2.704414e-32 | 58 | 10.8127837 | TRUE | 1 | NA | NA | NA | 2.694814e-29 | 1 | <!----> |\n",
       "| 21 | TIPARP    | 1.034107e-31 | 58 | 16.4674615 | TRUE | 1 | NA | NA | NA | 9.813674e-29 | 1 | <!----> |\n",
       "| 22 | AGTR2     | 2.877050e-31 | 58 |  8.5752236 | TRUE | 1 | NA | NA | NA | 2.606215e-28 | 1 | <!----> |\n",
       "| 23 | MTRNR2L6  | 3.881015e-30 | 22 |  0.6489617 | TRUE | 1 | NA | NA | NA | 3.362815e-27 | 1 | <!----> |\n",
       "| 24 | FOXR2     | 8.128059e-30 | 38 |  3.1681834 | TRUE | 1 | NA | NA | NA | 6.749337e-27 | 1 | <!----> |\n",
       "| 25 | NPBWR1    | 8.955406e-30 | 58 |  9.3804133 | TRUE | 1 | NA | NA | NA | 7.138892e-27 | 1 | <!----> |\n",
       "| 26 | BTG1      | 1.079333e-29 | 57 |  9.7249912 | TRUE | 1 | NA | NA | NA | 8.273084e-27 | 1 | <!----> |\n",
       "| 27 | MAGEC2    | 5.106952e-29 | 49 |  6.5372291 | TRUE | 1 | NA | NA | NA | 3.769498e-26 | 1 | <!----> |\n",
       "| 28 | MTRNR2L12 | 5.333579e-29 | 49 |  6.0658844 | TRUE | 1 | NA | NA | NA | 3.796175e-26 | 1 | <!----> |\n",
       "| 29 | SLITRK3   | 7.334380e-29 | 58 | 12.3594610 | TRUE | 1 | NA | NA | NA | 5.040236e-26 | 1 | <!----> |\n",
       "| 30 | CEBPA     | 8.781243e-29 | 52 |  8.6121095 | TRUE | 1 | NA | NA | NA | 5.833380e-26 | 1 | <!----> |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 71 | SDHAF3   | 8.052481e-22 | 58 | 23.747352 | TRUE | 1 | NA | NA | NA | 2.280830e-19 | 1 | <!----> |\n",
       "| 72 | CSTF2T   | 9.197873e-22 | 40 |  6.198350 | TRUE | 1 | NA | NA | NA | 2.545895e-19 | 1 | <!----> |\n",
       "| 73 | PTGS2    | 9.921123e-22 | 58 | 14.252650 | TRUE | 1 | NA | NA | NA | 2.708467e-19 | 1 | <!----> |\n",
       "| 74 | GPX6     | 1.040557e-21 | 52 | 14.488189 | TRUE | 1 | NA | NA | NA | 2.802333e-19 | 1 | <!----> |\n",
       "| 75 | KCNA5    | 1.057900e-21 | 49 |  9.525230 | TRUE | 1 | NA | NA | NA | 2.811051e-19 | 1 | <!----> |\n",
       "| 76 | ZXDB     | 2.205257e-21 | 40 |  7.520052 | TRUE | 1 | NA | NA | NA | 5.782705e-19 | 1 | <!----> |\n",
       "| 77 | FOXB2    | 2.351991e-21 | 33 |  4.287914 | TRUE | 1 | NA | NA | NA | 6.087380e-19 | 1 | <!----> |\n",
       "| 78 | HDGFL1   | 2.531268e-21 | 46 |  9.643443 | TRUE | 1 | NA | NA | NA | 6.467390e-19 | 1 | <!----> |\n",
       "| 79 | NEUROD6  | 7.452465e-21 | 47 |  8.717012 | TRUE | 1 | NA | NA | NA | 1.880002e-18 | 1 | <!----> |\n",
       "| 80 | RPRM     | 7.885459e-21 | 36 |  5.568092 | TRUE | 1 | NA | NA | NA | 1.964366e-18 | 1 | <!----> |\n",
       "| 81 | SOX1     | 1.056835e-20 | 42 |  7.776263 | TRUE | 1 | NA | NA | NA | 2.600204e-18 | 1 | <!----> |\n",
       "| 82 | TAS2R16  | 1.463009e-20 | 22 |  1.332422 | TRUE | 1 | NA | NA | NA | 3.555646e-18 | 1 | <!----> |\n",
       "| 83 | DUX4     | 1.691051e-20 | 57 | 24.840549 | TRUE | 1 | NA | NA | NA | 4.060355e-18 | 1 | <!----> |\n",
       "| 84 | PCDH10   | 1.758555e-20 | 58 | 23.111651 | TRUE | 1 | NA | NA | NA | 4.172172e-18 | 1 | <!----> |\n",
       "| 85 | VCX2     | 1.783939e-20 | 38 |  5.687692 | TRUE | 1 | NA | NA | NA | 4.182602e-18 | 1 | <!----> |\n",
       "| 86 | RAB6C    | 2.096076e-20 | 55 | 12.277323 | TRUE | 1 | NA | NA | NA | 4.857289e-18 | 1 | <!----> |\n",
       "| 87 | SLC6A15  | 2.425812e-20 | 58 | 22.830201 | TRUE | 1 | NA | NA | NA | 5.556783e-18 | 1 | <!----> |\n",
       "| 88 | HAND2    | 2.653981e-20 | 57 | 14.147082 | TRUE | 1 | NA | NA | NA | 5.942831e-18 | 1 | <!----> |\n",
       "| 89 | SERPINA7 | 2.632022e-20 | 44 |  8.436398 | TRUE | 1 | NA | NA | NA | 5.942831e-18 | 1 | <!----> |\n",
       "| 90 | HSPA13   | 5.288513e-20 | 58 | 15.913189 | TRUE | 1 | NA | NA | NA | 1.171053e-17 | 1 | <!----> |\n",
       "| 91 | TMEM229A | 5.544651e-20 | 48 | 10.625976 | TRUE | 1 | NA | NA | NA | 1.214279e-17 | 1 | <!----> |\n",
       "| 92 | OR4C5    | 6.026160e-20 | 34 |  4.290414 | TRUE | 1 | NA | NA | NA | 1.305384e-17 | 1 | <!----> |\n",
       "| 93 | DDIT4L   | 6.229151e-20 | 56 | 13.453267 | TRUE | 1 | NA | NA | NA | 1.334847e-17 | 1 | <!----> |\n",
       "| 94 | ZBTB18   | 8.589195e-20 | 55 | 14.259962 | TRUE | 1 | NA | NA | NA | 1.821001e-17 | 1 | <!----> |\n",
       "| 95 | OR4C13   | 9.601452e-20 | 31 |  3.509284 | TRUE | 1 | NA | NA | NA | 2.014182e-17 | 1 | <!----> |\n",
       "| 96 | ITM2A    | 1.137988e-19 | 54 | 13.179017 | TRUE | 1 | NA | NA | NA | 2.362391e-17 | 1 | <!----> |\n",
       "| 97 | AVPR1A   | 1.279005e-19 | 56 | 15.392623 | TRUE | 1 | NA | NA | NA | 2.627763e-17 | 1 | <!----> |\n",
       "| 98 | CAVIN4   | 1.582745e-19 | 55 | 13.976276 | TRUE | 1 | NA | NA | NA | 3.218626e-17 | 1 | <!----> |\n",
       "| 99 | ADRA2C   | 3.445722e-19 | 52 | 12.225444 | TRUE | 1 | NA | NA | NA | 6.936342e-17 | 1 | <!----> |\n",
       "| 100 | TCF21    | 3.793826e-19 | 58 | 15.381852 | TRUE | 1 | NA | NA | NA | 7.560716e-17 | 1 | <!----> |\n",
       "\n"
      ],
      "text/plain": [
       "    id        pp_element   element_muts_obs element_muts_exp element_enriched\n",
       "1   PABPC1L2A 1.396447e-74 45                1.0346828       TRUE            \n",
       "2   CT47B1    5.499257e-40 35                3.0487297       TRUE            \n",
       "3   PRR32     3.949606e-39 49                3.9203678       TRUE            \n",
       "4   FAM47C    7.746220e-39 50                5.3386166       TRUE            \n",
       "5   TSPYL5    3.546338e-38 57                6.8420465       TRUE            \n",
       "6   FABP5     8.191851e-38 58                7.7089385       TRUE            \n",
       "7   OR4A47    1.238507e-37 38                1.9170070       TRUE            \n",
       "8   PABPC4L   9.526415e-36 57                7.2895514       TRUE            \n",
       "9   GYPB      1.191023e-35 55               11.1995994       TRUE            \n",
       "10  DIO3      1.932920e-35 44                3.9130694       TRUE            \n",
       "11  TMEM215   2.577214e-35 57                7.5837952       TRUE            \n",
       "12  REG3G     6.070298e-35 51                5.4927299       TRUE            \n",
       "13  GYPA      8.332016e-34 58               14.2045127       TRUE            \n",
       "14  RTL3      1.225487e-33 49                5.3182638       TRUE            \n",
       "15  C10orf142 8.768327e-33 50                6.2485105       TRUE            \n",
       "16  ZNF66     1.472389e-32 58               14.9259388       TRUE            \n",
       "17  CXCR4     1.716483e-32 49                7.4614960       TRUE            \n",
       "18  PABPC1L2B 1.848870e-32 23                0.9542834       TRUE            \n",
       "19  MTRNR2L13 2.127050e-32 58                7.7674464       TRUE            \n",
       "20  MARCKS    2.704414e-32 58               10.8127837       TRUE            \n",
       "21  TIPARP    1.034107e-31 58               16.4674615       TRUE            \n",
       "22  AGTR2     2.877050e-31 58                8.5752236       TRUE            \n",
       "23  MTRNR2L6  3.881015e-30 22                0.6489617       TRUE            \n",
       "24  FOXR2     8.128059e-30 38                3.1681834       TRUE            \n",
       "25  NPBWR1    8.955406e-30 58                9.3804133       TRUE            \n",
       "26  BTG1      1.079333e-29 57                9.7249912       TRUE            \n",
       "27  MAGEC2    5.106952e-29 49                6.5372291       TRUE            \n",
       "28  MTRNR2L12 5.333579e-29 49                6.0658844       TRUE            \n",
       "29  SLITRK3   7.334380e-29 58               12.3594610       TRUE            \n",
       "30  CEBPA     8.781243e-29 52                8.6121095       TRUE            \n",
       "⋮   ⋮         ⋮            ⋮                ⋮                ⋮               \n",
       "71  SDHAF3    8.052481e-22 58               23.747352        TRUE            \n",
       "72  CSTF2T    9.197873e-22 40                6.198350        TRUE            \n",
       "73  PTGS2     9.921123e-22 58               14.252650        TRUE            \n",
       "74  GPX6      1.040557e-21 52               14.488189        TRUE            \n",
       "75  KCNA5     1.057900e-21 49                9.525230        TRUE            \n",
       "76  ZXDB      2.205257e-21 40                7.520052        TRUE            \n",
       "77  FOXB2     2.351991e-21 33                4.287914        TRUE            \n",
       "78  HDGFL1    2.531268e-21 46                9.643443        TRUE            \n",
       "79  NEUROD6   7.452465e-21 47                8.717012        TRUE            \n",
       "80  RPRM      7.885459e-21 36                5.568092        TRUE            \n",
       "81  SOX1      1.056835e-20 42                7.776263        TRUE            \n",
       "82  TAS2R16   1.463009e-20 22                1.332422        TRUE            \n",
       "83  DUX4      1.691051e-20 57               24.840549        TRUE            \n",
       "84  PCDH10    1.758555e-20 58               23.111651        TRUE            \n",
       "85  VCX2      1.783939e-20 38                5.687692        TRUE            \n",
       "86  RAB6C     2.096076e-20 55               12.277323        TRUE            \n",
       "87  SLC6A15   2.425812e-20 58               22.830201        TRUE            \n",
       "88  HAND2     2.653981e-20 57               14.147082        TRUE            \n",
       "89  SERPINA7  2.632022e-20 44                8.436398        TRUE            \n",
       "90  HSPA13    5.288513e-20 58               15.913189        TRUE            \n",
       "91  TMEM229A  5.544651e-20 48               10.625976        TRUE            \n",
       "92  OR4C5     6.026160e-20 34                4.290414        TRUE            \n",
       "93  DDIT4L    6.229151e-20 56               13.453267        TRUE            \n",
       "94  ZBTB18    8.589195e-20 55               14.259962        TRUE            \n",
       "95  OR4C13    9.601452e-20 31                3.509284        TRUE            \n",
       "96  ITM2A     1.137988e-19 54               13.179017        TRUE            \n",
       "97  AVPR1A    1.279005e-19 56               15.392623        TRUE            \n",
       "98  CAVIN4    1.582745e-19 55               13.976276        TRUE            \n",
       "99  ADRA2C    3.445722e-19 52               12.225444        TRUE            \n",
       "100 TCF21     3.793826e-19 58               15.381852        TRUE            \n",
       "    pp_site site_muts_obs site_muts_exp site_enriched fdr_element  fdr_site\n",
       "1   1       NA            NA            NA            2.782979e-70 1       \n",
       "2   1       NA            NA            NA            5.479734e-36 1       \n",
       "3   1       NA            NA            NA            2.623723e-35 1       \n",
       "4   1       NA            NA            NA            3.859361e-35 1       \n",
       "5   1       NA            NA            NA            1.413499e-34 1       \n",
       "6   1       NA            NA            NA            2.720923e-34 1       \n",
       "7   1       NA            NA            NA            3.526031e-34 1       \n",
       "8   1       NA            NA            NA            2.373149e-32 1       \n",
       "9   1       NA            NA            NA            2.637323e-32 1       \n",
       "10  1       NA            NA            NA            3.852115e-32 1       \n",
       "11  1       NA            NA            NA            4.669209e-32 1       \n",
       "12  1       NA            NA            NA            1.008125e-31 1       \n",
       "13  1       NA            NA            NA            1.277298e-30 1       \n",
       "14  1       NA            NA            NA            1.744480e-30 1       \n",
       "15  1       NA            NA            NA            1.164960e-29 1       \n",
       "16  1       NA            NA            NA            1.833952e-29 1       \n",
       "17  1       NA            NA            NA            2.012223e-29 1       \n",
       "18  1       NA            NA            NA            2.047008e-29 1       \n",
       "19  1       NA            NA            NA            2.231051e-29 1       \n",
       "20  1       NA            NA            NA            2.694814e-29 1       \n",
       "21  1       NA            NA            NA            9.813674e-29 1       \n",
       "22  1       NA            NA            NA            2.606215e-28 1       \n",
       "23  1       NA            NA            NA            3.362815e-27 1       \n",
       "24  1       NA            NA            NA            6.749337e-27 1       \n",
       "25  1       NA            NA            NA            7.138892e-27 1       \n",
       "26  1       NA            NA            NA            8.273084e-27 1       \n",
       "27  1       NA            NA            NA            3.769498e-26 1       \n",
       "28  1       NA            NA            NA            3.796175e-26 1       \n",
       "29  1       NA            NA            NA            5.040236e-26 1       \n",
       "30  1       NA            NA            NA            5.833380e-26 1       \n",
       "⋮   ⋮       ⋮             ⋮             ⋮             ⋮            ⋮       \n",
       "71  1       NA            NA            NA            2.280830e-19 1       \n",
       "72  1       NA            NA            NA            2.545895e-19 1       \n",
       "73  1       NA            NA            NA            2.708467e-19 1       \n",
       "74  1       NA            NA            NA            2.802333e-19 1       \n",
       "75  1       NA            NA            NA            2.811051e-19 1       \n",
       "76  1       NA            NA            NA            5.782705e-19 1       \n",
       "77  1       NA            NA            NA            6.087380e-19 1       \n",
       "78  1       NA            NA            NA            6.467390e-19 1       \n",
       "79  1       NA            NA            NA            1.880002e-18 1       \n",
       "80  1       NA            NA            NA            1.964366e-18 1       \n",
       "81  1       NA            NA            NA            2.600204e-18 1       \n",
       "82  1       NA            NA            NA            3.555646e-18 1       \n",
       "83  1       NA            NA            NA            4.060355e-18 1       \n",
       "84  1       NA            NA            NA            4.172172e-18 1       \n",
       "85  1       NA            NA            NA            4.182602e-18 1       \n",
       "86  1       NA            NA            NA            4.857289e-18 1       \n",
       "87  1       NA            NA            NA            5.556783e-18 1       \n",
       "88  1       NA            NA            NA            5.942831e-18 1       \n",
       "89  1       NA            NA            NA            5.942831e-18 1       \n",
       "90  1       NA            NA            NA            1.171053e-17 1       \n",
       "91  1       NA            NA            NA            1.214279e-17 1       \n",
       "92  1       NA            NA            NA            1.305384e-17 1       \n",
       "93  1       NA            NA            NA            1.334847e-17 1       \n",
       "94  1       NA            NA            NA            1.821001e-17 1       \n",
       "95  1       NA            NA            NA            2.014182e-17 1       \n",
       "96  1       NA            NA            NA            2.362391e-17 1       \n",
       "97  1       NA            NA            NA            2.627763e-17 1       \n",
       "98  1       NA            NA            NA            3.218626e-17 1       \n",
       "99  1       NA            NA            NA            6.936342e-17 1       \n",
       "100 1       NA            NA            NA            7.560716e-17 1       \n",
       "    has_site_mutations\n",
       "1                     \n",
       "2                     \n",
       "3                     \n",
       "4                     \n",
       "5                     \n",
       "6                     \n",
       "7                     \n",
       "8                     \n",
       "9                     \n",
       "10                    \n",
       "11                    \n",
       "12                    \n",
       "13                    \n",
       "14                    \n",
       "15                    \n",
       "16                    \n",
       "17                    \n",
       "18                    \n",
       "19                    \n",
       "20                    \n",
       "21                    \n",
       "22                    \n",
       "23                    \n",
       "24                    \n",
       "25                    \n",
       "26                    \n",
       "27                    \n",
       "28                    \n",
       "29                    \n",
       "30                    \n",
       "⋮   ⋮                 \n",
       "71                    \n",
       "72                    \n",
       "73                    \n",
       "74                    \n",
       "75                    \n",
       "76                    \n",
       "77                    \n",
       "78                    \n",
       "79                    \n",
       "80                    \n",
       "81                    \n",
       "82                    \n",
       "83                    \n",
       "84                    \n",
       "85                    \n",
       "86                    \n",
       "87                    \n",
       "88                    \n",
       "89                    \n",
       "90                    \n",
       "91                    \n",
       "92                    \n",
       "93                    \n",
       "94                    \n",
       "95                    \n",
       "96                    \n",
       "97                    \n",
       "98                    \n",
       "99                    \n",
       "100                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(results, 100L)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 remove hypermut, n= 0 ,  0 %\n",
      "hypermuted samples:   \n",
      "\n",
      "reversing 0 positions\n",
      "Removing  0  invalid SNVs & indels\n",
      "\n",
      "Number of Elements with 0 Mutations:  16359 \n",
      "Tests to do:  53868 \n"
     ]
    }
   ],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/WGS/ICGC/daragh_icgc_brca_cohort_mutations_icgc_3.non_coding.tsv\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(ref)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(alt)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(ref) != \"-\")\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(alt) != \"-\")\n",
    "\n",
    "#Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/TN_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "\n",
    "\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 40,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/TN_regs_ICGC.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 remove hypermut, n= 0 ,  0 %\n",
      "hypermuted samples:   \n",
      "\n",
      "reversing 0 positions\n",
      "Removing  0  invalid SNVs & indels\n",
      "\n",
      "Number of Elements with 0 Mutations:  7400 \n",
      "Tests to do:  30807 \n"
     ]
    }
   ],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/WGS/ICGC/daragh_icgc_brca_cohort_mutations_icgc_3.non_coding.tsv\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(ref)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(alt)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(ref) != \"-\")\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(alt) != \"-\")\n",
    "\n",
    "#Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/Primary_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 24,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/Primary_regs_ICGC.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 remove hypermut, n= 0 ,  0 %\n",
      "hypermuted samples:   \n",
      "\n",
      "reversing 0 positions\n",
      "Removing  0  invalid SNVs & indels\n",
      "\n",
      "Number of Elements with 0 Mutations:  3992 \n",
      "Tests to do:  15487 \n"
     ]
    }
   ],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/WGS/ICGC/daragh_icgc_brca_cohort_mutations_icgc_3.non_coding.tsv\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(ref)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(alt)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(ref) != \"-\")\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(alt) != \"-\")\n",
    "\n",
    "#Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/ER_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 24,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/ER_regs_ICGC.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/WGS/ICGC/daragh_icgc_brca_cohort_mutations_icgc_3.non_coding.tsv\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(ref)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(alt)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(ref) != \"-\")\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(alt) != \"-\")\n",
    "\n",
    "#Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/TN_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 24,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/TN_regs_ICGC.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“multiple methods tables found for ‘aperm’”\n",
      "Warning message:\n",
      "“replacing previous import ‘BiocGenerics::aperm’ by ‘DelayedArray::aperm’ when loading ‘SummarizedExperiment’”\n",
      "Warning message:\n",
      "“no function found corresponding to methods exports from ‘BSgenome’ for: ‘releaseName’”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 remove hypermut, n= 0 ,  0 %\n",
      "hypermuted samples:   \n",
      "\n",
      "reversing 0 positions\n",
      "Removing  0  invalid SNVs & indels\n",
      "\n",
      "Number of Elements with 0 Mutations:  14590 \n",
      "Tests to do:  23617 \n"
     ]
    }
   ],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/CADD/Hartwig/Hartwig_high_CADD.minimal.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(ref)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(alt)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(ref) != \"-\")\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(alt) != \"-\")\n",
    "\n",
    "#Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/Primary_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 24,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/Primary_regs_Hartwig_CADD.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 remove hypermut, n= 0 ,  0 %\n",
      "hypermuted samples:   \n",
      "\n",
      "reversing 0 positions\n",
      "Removing  0  invalid SNVs & indels\n",
      "\n",
      "Number of Elements with 0 Mutations:  5002 \n",
      "Tests to do:  14477 \n"
     ]
    }
   ],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/CADD/Hartwig/Hartwig_high_CADD.minimal.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(ref)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(alt)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(ref) != \"-\")\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(alt) != \"-\")\n",
    "\n",
    "#Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/ER_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 24,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/ER_regs_Hartwig_CADD.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 remove hypermut, n= 0 ,  0 %\n",
      "hypermuted samples:   \n",
      "\n",
      "reversing 0 positions\n",
      "Removing  0  invalid SNVs & indels\n",
      "\n",
      "Number of Elements with 0 Mutations:  32732 \n",
      "Tests to do:  32051 \n"
     ]
    }
   ],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/CADD/Hartwig/Hartwig_high_CADD.minimal.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(ref)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(alt)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(ref) != \"-\")\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(alt) != \"-\")\n",
    "\n",
    "#Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/TN_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 24,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/TN_regs_Hartwig_CADD.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "#Mutations\n",
    "\n",
    "library(ActiveDriverWGS)\n",
    "\n",
    "Hartwig_all_non_coding = read.table(file=\"/home/darragh/New/CADD/ICGC/ICGC_high_CADD.minimal.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Hartwig_all_non_coding) = c(\"chr\", \"pos1\", \"pos2\", \"ref\", \"alt\" ,\"patient\")\n",
    "Hartwig_all_non_coding = Hartwig_all_non_coding[Hartwig_all_non_coding$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(ref)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, nchar(as.character(alt)) < 2)\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(ref) != \"-\")\n",
    "Hartwig_all_non_coding = subset(Hartwig_all_non_coding, as.character(alt) != \"-\")\n",
    "\n",
    "#Regions\n",
    "Primary_regs = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/Primary_regs.txt\",header = FALSE, sep = \"\\t\")\n",
    "colnames(Primary_regs) = c(\"chr\", \"start\", \"end\", \"id\")\n",
    "Primary_regs = Primary_regs[Primary_regs$chr %in% c(\"chr1\", \"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\"chr18\",\"chr19\",\"chr20\",\"chr21\",\"chr22\",\"chrX\",\"chrY\" ), ]\n",
    "\n",
    "results = ActiveDriverWGS(mutations = Hartwig_all_non_coding,\n",
    "                          elements = Primary_regs,\n",
    "                          window_size = 50000,\n",
    "                          filter_hyper_MB = 3000,\n",
    "                          mc.cores = 24,\n",
    "                          ref_genome = \"hg19\")\n",
    "\n",
    "write.table(results, \"/home/darragh/New/ActiveDriverWGS/output/Primary_regs_ICGC_CADD.txt\",row.names = F, col.names =F, sep = \"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "Muts = read.table(file=\"/home/darragh/New/ActiveDriverWGS/inputs/Hartwig_non_coding.txt\",header = T, sep = \"\\t\")\n",
    "head(Muts)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
